xdsl/utils/dialect_codegen.py

```diff
@@ -22,6 +22,7 @@
     VarOperandDef,
     VarResultDef,
 )
+from xdsl.utils.runtime_final import runtime_final
```

Comment:
seems unrelated?

Reply:
`BaseAttr` uses the runtime final check to determine what's a `base` is.